src/app/core/public/api/impl/Circuit.ts

```diff
@@ -155,7 +159,23 @@ export abstract class CircuitImpl implements Circuit {
         //  For now, ignore the `space`, and ignore any non-Component
         //   objects that are selected
         //  From these components, average their positions
-        throw new Error("Method not implemented.");
+        const allComponents = this.selections.get();
+
+        // Case: no components are selected
+        if (allComponents.length === 0) {
+            return V(0,0)
+        }
+
+        // Case: One or more components are selected
+        let comps =  allComponents.map((id) => this.getComponent(id))
+                    .filter((comp) => (comp !== undefined)) as Component[];
+        let sumPositions = comps.reduce((vec1, vec2) => V(vec1.x, vec1.y).add(vec2.pos.x, vec2?.pos.y), V(0,0));
```

Comment:
`Vector` handily already provides an easy way to just add two vectors, so you can just do
`(vec1, vec2) => vec1.add(vec2.pos)`

Reply:
I would also just 1-line this as:

```ts
const sumPosition = allComponents
    .map((c) => c.pos)
    .reduce((sum, v) => sum.add(v));
```